aws-api/src/main/java/com/amplifyframework/api/aws/AWSApiPlugin.java

```diff
@@ -600,7 +630,28 @@ private String selectApiName(Set<String> apiClients) throws ApiException {
             );
         }
 
-        return AppSyncGraphQLOperation.<R>builder()
+        AuthModeStrategyType authModeStrategyType = AuthModeStrategyType.DEFAULT;
+        // If it's an AppSyncGraphQLRequest AND
+        // No authorizationType is set on the request AND
+        // authModeStrategyType is set on the request.
+        if (graphQLRequest instanceof AppSyncGraphQLRequest<?> &&
+            ((AppSyncGraphQLRequest<?>) graphQLRequest).getAuthorizationType() == null &&
+            ((AppSyncGraphQLRequest<?>) graphQLRequest).getAuthModeStrategyType() != null) {
+            authModeStrategyType = ((AppSyncGraphQLRequest<?>) graphQLRequest).getAuthModeStrategyType();
+        }
+        if (AuthModeStrategyType.MULTIAUTH.equals(authModeStrategyType)) {
```

Comment:
I can change this one :) I think the other one I commented on has a body that's a little longer so it makes it look weird. This one is pretty compact so it looks OK.

Reply:
I actually did a little cleanup. The switch just didn't look right. It got flagged by the IDE because it only has a one case and the default branch. Turns out I didn't really need the else branch anyways.